### ISSUE TITLE
Install CloudComPy with pip in a virtual environment

### ISSUE THREAD

**Issue by ttsesm** [CloudCompare/CloudComPy#13]
Hi Guys,

Do you know whether it is possible to install CloudCompPy and have a python binding with pip in a virtual environment? My plan is to compile it from source and then possibly getting a `setup.py` should do the trick.

However, I am not sure whether it would work or you have tried to install the lib that way.

Thanks.

**Comment by prascle** [comment]
Hello,
Compiling CloudComPy with a setup.py and using pip is indeed something I would like to do. I haven't tried it yet, it's an area where I still have a lot to learn.
It's probably doable, but I'd start by limiting the CloudCompare plugins to the bare minimum.
Depending on the platform, there are compatibility problems between the different prerequisites or with CloudCompare that can result in compilation problems or bugs at runtime. This is what I have noticed in my different builds on Windows and Linux.
I intend to increase my skills on this subject and if you already have experience in packaging with pip, your contribution is welcome. 
Regards,
Paul

**Comment by ttsesm** [comment]
Hi Paul,

Thanks for the feedback. To be honest, I do not have a direct experience with something like that as well. However, looking on some [other projects](https://teaser.readthedocs.io/en/latest/installation.html#installing-python-bindings) that I was working with lately and how they do it, I guess I would follow a similar approach and possibly go with [pybind11](https://github.com/pybind/pybind11) as well. It seems a straight forward approach.

I guess then you need to modify accordingly the CMakelists.txt file and possibly to do some [binding to the C++ functions](https://github.com/MIT-SPARK/TEASER-plusplus/blob/master/python/teaserpp_python/teaserpp_python.cc) or something similar and it should work.

**Comment by prascle** [comment]
Yes, switching to pybind11 instead of Boost.Python is probably a step that will make pip packaging easier. This is one of the things I need to look into. I don't have a clear idea of how much effort goes into adapting or rewriting the interface, but I think it's a pretty systematic transposition, pybind11 started from Boost.Python. 
The impact on the CMakeList.txt should be minimal. The C++ Python binding is indeed the object of CloudComPy, I've been driving it manually so far to keep the Python interface fairly simple, matching the CloudCompare command language, and I'm enriching the interface as needed.

**Comment by s-du** [comment]
Hi, I was wondering if there is any further feedback regarding this. I would really love to install CloudCompy in a venv, without needing Anaconda. Does anyone knows if it is achievable?
